src/mudlet-lua/lua/Other.lua

```diff
@@ -832,3 +845,51 @@ do
     end
   end
 end
+
+
+-- A utility function based on tempTimers meant to track all sorts of temporary states
+local timeframeTable = {}  -- keeps track of all timeframe timers
+function timeframe(vname, true_time, nil_time, ...)
+  local timerlist = {
+    { 0 }, -- reset variable 'vname' to nil
+    true_time and { true_time, true }, -- when to set variable to true
+    nil_time and { nil_time }, -- when to set variable back to nil
+    ...  -- further list of times and values to set the variable to
+  }
+
+  -- reset any associated timeframes for 'vname'
+  killTimeframe(vname)
+  timeframeTable[vname] = {}
+
+  -- run through timerlist data and create tempTimers to set 'vname' variable data
+  local maxtime = 0
+  for step, data in ipairs(timerlist) do
+    local time = data[1]
+    local value = data[2]
+    maxtime = (time > maxtime) and time or maxtime
+    if time <= 0 then
+      assert(loadstring(format("%s = %s", vname, tostring(value or nil))))()
```

Comment:
This looks buggy in multiple ways:
- If `value` is a table, this will probably fail
- if `value` is `false`, it will become `nil`
- The `or nil` is useless in general. That branch is only reached if `value` is `nil` anyways or `false` (see above)
- If `value` is a string (like `foo`), the code will look like `var = foo`, which will assign the value of the global variable `foo` (most likely nil). If the string contains spaces, it will most likely result in a syntax error. In the `string` case, you'll need to surround the right-hand value by quotes.
- I see no localization of `string.format` as `format`

Another not-so-nice thing is that both `if` branches contain the same `assert` line. You may be able to use the anonymous function in the `else` branch in both places.